includes/core/server/request/class-list-charge-refunds.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\List_Charge_Refunds.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Core\Server\Request;
+use WC_Payments_API_Client;
+
+/**
+ * Request class for charge refunds.
+ */
+class List_Charge_Refunds extends Request {
+
+	const DEFAULT_PARAMS = [
+		'limit' => 100,
+	];
+
+	const IMMUTABLE_PARAMS = [ 'id' ];
+
+	/**
+	 * Sets the charge ID, which will be used in the request URL.
+	 *
+	 * @param string $id Sets the charge ID, which will be used in the request URL.
+	 *
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	protected function set_id( string $id ) {
+		/**
+		 * `py_XYZ` objects are identical to charges, and sometimes occur
+		 * whenever the payment was made in a non-deposit currency.
+		 */
+		$this->validate_stripe_id( $id, [ 'ch', 'py' ] );
+		$this->id = $id;
+	}
+
+	/**
+	 * Stores the limit for the lists.
+	 *
+	 * @param int $limit Number to limit results.
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function set_limit( int $limit ) {
+		$this->validate_is_larger_then( $limit, 0 );
+		$this->set_param( 'limit', $limit );
+	}
+
+	/**
+	 * Returns the request's API.
+	 *
+	 * @return string
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function get_api(): string {
+		return WC_Payments_API_Client::REFUNDS_API . '/' . $this->id;
```

Comment:
The current E2E failures such as [this one](https://github.com/Automattic/woocommerce-payments/actions/runs/4656337625/jobs/8239863085?pr=5967) are related to this wrong endpoint. Might check this again.

The correct endpoint would be `WC_Payments_API_Client::REFUNDS_API` with the provided parameter `charge`
https://github.com/Automattic/woocommerce-payments/blob/827bb80c14c531c5d6b2ec52745a7fb002108b3c/includes/wc-payment-api/class-wc-payments-api-client.php#L222-L228

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/5967/commits/9b6b1e4cade9e4deb0f402fa44872f9e6890cd43